exporter/sumologicexporter/exporter.go

```diff
@@ -178,3 +178,101 @@ func (se *sumologicexporter) pushLogsData(_ context.Context, ld pdata.Logs) (int
 
 	return 0, nil
 }
+
+func newMetricsExporter(
+	cfg *Config,
+	params component.ExporterCreateParams,
+) (component.MetricsExporter, error) {
+	se, err := initExporter(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return exporterhelper.NewMetricsExporter(
+		cfg,
+		params.Logger,
+		se.pushMetricsData,
+		// Disable exporterhelper Timeout, since we are using a custom mechanism
+		// within exporter itself
+		exporterhelper.WithTimeout(exporterhelper.TimeoutSettings{Timeout: 0}),
+		exporterhelper.WithRetry(cfg.RetrySettings),
+		exporterhelper.WithQueue(cfg.QueueSettings),
+	)
+}
+
+// pushMetricsData groups data with common metadata and send them as separate batched requests
+// it returns number of unsent metrics and error which contains list of dropped records
+// so they can be handle by the OTC retry mechanism
+func (se *sumologicexporter) pushMetricsData(_ context.Context, ld pdata.Metrics) (int, error) {
```

Comment:
Can we mark those unused contexts with TODO comments?
Otelcol community is fine with such an approach?

Reply:
What do you want to put into those TODO?

Reply:
> contextify this method

:) Ideally this should be used when issuing requests. Is it done this way in other exporters?

Reply:
It's common approach across the exporters

Reply:
but seems that we will have to pass it to the sender anyway